### PR TITLE
304 feature utility to communicate to engine api

### PR DESCRIPTION
**Pull request by arminsabouri**
@scottmillner Drafting until I refactor tasks to use these.
Wanted to put up these commits for visibility on the interface

### CODE REVIEW

File: crates/consensus/authority/src/engine_util.rs
Comment (arminsabouri) [inline crates/consensus/authority/src/engine_util.rs:95]: Note for reviewers: there are two things we can do when the engine API isnt responding

If they are offline (the channel cannot be open) -> Should we panic?
What happens if they are syncing for a long time -> panic? wait? timeout?

Comment (scottmillner) [reply crates/consensus/authority/src/engine_util.rs:95]: i think we should panic if the channel can't be opened bc that seems critical and no point continuing until that is resolved

syncing for a long time: this could be for different reasons (unstable peers to download headers from, connectivity issues) so maybe timeout and retry then panic over retry threshold

**Comment by arminsabouri** [comment]
Removed the last commit which changes the authority task. There are currently too many PRs that change that file. I will update once #76 is merged

**Review by scottmillner — APPROVED** [review]

File: crates/consensus/authority/src/engine_util.rs
Comment (stevenroose) [inline crates/consensus/authority/src/engine_util.rs:126]: 500ms is a _really_ long time in CPU world, though.. This could easily be 50.

Also, you can't use std::thread::sleep in async functions, that will actually hold up the runtime thread. Should use `tokio::time::sleep().await`.

OK, I see these two things are already addressed ;) good job :)

**Pull request merged**
